tests/Unit/DataStructures/Test_Index.cpp

```diff
@@ -16,6 +16,20 @@
 #include "Utilities/Literals.hpp"
 #include "tests/Unit/TestHelpers.hpp"
 
+namespace {
+template <size_t Dim>
+void test_collapsed_index(const Index<Dim> extents) noexcept {
+  for (IndexIterator<Dim> index_it(extents); index_it; ++index_it) {
+    CAPTURE(*index_it);
+    std::array<size_t, Dim> index;
+    for (size_t d = 0; d < Dim; ++d) {
+      index[d] = (*index_it)[d];
+    }
+    CHECK(index_it.collapsed_index() == extents.collapsed_index(index));
```

Comment:
Please add comment like "Compare to the collapsed_index that is computed independently by the IndexIterator during iteration" to make clear this test is not redundant